packages/amplify-frontend-javascript/lib/frontend-config-creator.js

```diff
@@ -274,29 +277,35 @@ function getAWSExportsObject(resources) {
 }
 
 async function getCurrentAWSExports(context) {
-  const { amplify } = context;
-  const projectPath = context.exeInfo ? context.exeInfo.localEnvInfo.projectPath : amplify.getEnvInfo().projectPath;
-  const projectConfig = context.exeInfo ? context.exeInfo.projectConfig[constants.Label] : amplify.getProjectConfig()[constants.Label];
-  const frontendConfig = projectConfig.config;
+  const { amplify, exeInfo } = context;
+  const projectPath = exeInfo?.localEnvInfo?.projectPath || amplify.getEnvInfo().projectPath;
+  const { config: frontendConfig } = exeInfo?.projectConfig?.[constants.Label] || amplify.getProjectConfig()[constants.Label];
   const srcDirPath = path.join(projectPath, frontendConfig.SourceDir);
 
   const targetFilePath = path.join(srcDirPath, constants.exportsFilename);
   let awsExports = {};
 
   if (fs.existsSync(targetFilePath)) {
-    // if packaged, we can't load an ES6 module because pkg doesn't support it yet
-    // eslint-disable-next-line spellcheck/spell-checker
-    const es5export = 'module.exports = {default: awsmobile};\n';
-    // eslint-disable-next-line spellcheck/spell-checker
-    const es6export = 'export default awsmobile;\n';
-
     const fileContents = fs.readFileSync(targetFilePath, 'utf-8');
-    fs.writeFileSync(targetFilePath, fileContents.replace(es6export, es5export));
-    // eslint-disable-next-line global-require, import/no-dynamic-require
-    awsExports = require(targetFilePath).default;
-    fs.writeFileSync(targetFilePath, fileContents);
+    try {
+      // transpile the file contents to CommonJS
+      const { code } = babel.transformSync(fileContents, {
+        plugins: [babelTransformEsmToCjs],
+      });
+      const m = new Module();
```

Comment:
nit: avoid using a single letter and use something like `module`

Reply:
`module` is a reserved keyword, is `mod` okay? or `temporaryModule`?

Reply:
Any of those work for me